api/v1alpha2/linodecluster_webhook_test.go

```diff
@@ -110,3 +111,80 @@ func TestValidateCreate(t *testing.T) {
 		),
 	)
 }
+
+func TestValidateDNSLinodeCluster(t *testing.T) {
+	t.Parallel()
+
+	var (
+		validCluster = LinodeCluster{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "example",
+				Namespace: "example",
+			},
+			Spec: LinodeClusterSpec{
+				Region: "us-ord",
+				Network: NetworkSpec{
+					LoadBalancerType:    "dns",
+					DNSRootDomain:       "test.net",
+					DNSUniqueIdentifier: "abc123",
+				},
+			},
+		}
+		noRootDomainCluster = LinodeCluster{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "example",
+				Namespace: "example",
+			},
+			Spec: LinodeClusterSpec{
+				Region: "us-ord",
+				Network: NetworkSpec{
+					LoadBalancerType:    "dns",
+					DNSRootDomain:       "",
+					DNSUniqueIdentifier: "abc123",
+				},
+			},
+		}
+		noUniqueIDCluster = LinodeCluster{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "example",
+				Namespace: "example",
+			},
+			Spec: LinodeClusterSpec{
+				Region: "us-ord",
+				Network: NetworkSpec{
+					LoadBalancerType:    "dns",
+					DNSRootDomain:       "test.net",
+					DNSUniqueIdentifier: "",
+				},
+			},
+		}
+	)
+
+	NewSuite(t, mock.MockLinodeClient{}).Run(
+		OneOf(
+			Path(
+				Call("valid", func(ctx context.Context, mck Mock) {
+					mck.LinodeClient.EXPECT().GetRegion(gomock.Any(), gomock.Any()).Return(nil, nil).AnyTimes()
+				}),
+				Result("success", func(ctx context.Context, mck Mock) {
+					assert.NoError(t, validCluster.validateLinodeCluster(ctx, mck.LinodeClient))
+				}),
+			),
+		),
+		OneOf(
+			Path(Call("no domain and unique id set", func(ctx context.Context, mck Mock) {
+				mck.LinodeClient.EXPECT().GetRegion(gomock.Any(), gomock.Any()).Return(nil, nil).AnyTimes()
+			})),
+		),
+		Result("error", func(ctx context.Context, mck Mock) {
+			require.EqualError(
+				t,
+				errors.New("LinodeCluster.infrastructure.cluster.x-k8s.io \"example\" is invalid: dnsRootDomain needs to be set when LoadBalancer Type is DNS: Required value"),
```

Comment:
Instead of looking for an exact match on a string we should probably check the error type is `field.ErrorTypeRequired` and that the error string contains "dnsRootDomain".

Reply:
done